### PR TITLE
Adding merge logic in fs-merge-tree

### PR DESCRIPTION
**Pull request by arthirm**

### CODE REVIEW

File: lib/fs-merge-tree.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 'use strict';
 
-const walkSync = require('walk-sync');
-
+var canSymlink = require('can-symlink')();
```
Comment (step2yeung) [inline lib/fs-merge-tree.js]: use `const` here?

File: lib/fs-merge-tree.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
 const FSTree = require('./');
+var defaultIsEqual = FSTree.defaultIsEqual;
```
Comment (step2yeung) [inline lib/fs-merge-tree.js]: use `let` here?

File: lib/index.js [hunk 3]
```diff
@@ -405,14 +405,12 @@ FSTree.prototype.match = function(globs) {
...
+  let patches;
+  this._ensureEntriesPopulated();
   if(this.srcTree ) {
-    this._ensureEntriesPopulated();
     var prevTree = new FSTree.fromEntries(this.prevEntries);
     patches = prevTree.calculatePatch(this);
-    this.prevEntries = [];
-    this.entries.forEach(e => this.prevEntries.push(e));
-
+    this.prevEntries = this.entries.slice();
```
Comment (step2yeung) [inline lib/index.js]: you probably need to rebase, since these changes has already landed :)

File: lib/fs-merge-tree.js (hunk 2 above)
Comment (step2yeung) [inline lib/fs-merge-tree.js]: Seeing a mixture of `var` and `let`, I assume the code that was copied over was all using var.. Shall we update this?

**Comment by stefanpenner** [comment]
This is great work, the resulting research lead to several improvements in the ecosystem. This PR itself though should likely be closed, as we will most likely not be actively pursuing this approach in the near term. Additionally the code-base has changed drastically, if we do explore this again it will look very different. We will still have this PR around for history, and context, but we closing it seems appropriate.

**Pull request closed without merging**